service/controller/v3/network/network.go

```diff
@@ -0,0 +1,74 @@
+package network
+
+import (
+	"context"
+	"net"
+
+	"github.com/giantswarm/ipam"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/azure-operator/service/controller/v3/key"
+)
+
+const (
+	masterSubnetMask = 24
+	workerSubnetMask = 24
+	vpnSubnetMask    = 24
+
+	ipv4MaskSize = 32
+)
+
+// ComputeFromCR computes subnets using network found in CR.
+func ComputeFromCR(ctx context.Context, obj interface{}) (*Subnets, error) {
+	azureConfig, err := key.ToCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	vnetCIDR := key.VnetCIDR(azureConfig)
+	_, vnet, err := net.ParseCIDR(vnetCIDR)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	subnets, err := Compute(*vnet)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return subnets, nil
+}
+
+// Compute computes subnets within network.
+//
+// subnets computation rely on ipam.Free and use ipv4MaskSize as IPMask length.
+func Compute(network net.IPNet) (subnets *Subnets, err error) {
```

Comment:
We don't need that public, do we?

Reply:
This is used in e2e test

https://github.com/giantswarm/azure-operator/pull/277/files#diff-8864153a0b634472800453cb4d1ad300L61

Reply:
Ack.